stacks-signer/src/stacks_client.rs

```diff
@@ -98,6 +172,173 @@ impl StacksClient {
+    ) -> Result<Vec<u8>, ClientError> {
+        self.seralize_sign_sponsored_tx_anchor_mode_version(
+            payload,
+            None,
+            sender_nonce,
+            None,
+            tx_fee,
+            anchor_mode,
+        )
+    }
+
+    fn seralize_sign_sponsored_tx_anchor_mode_version(
+        &self,
+        payload: TransactionPayload,
+        payer: Option<&StacksPrivateKey>,
+        sender_nonce: u64,
+        payer_nonce: Option<u64>,
+        tx_fee: u64,
+        anchor_mode: TransactionAnchorMode,
+    ) -> Result<Vec<u8>, ClientError> {
+        let mut sender_spending_condition = TransactionSpendingCondition::new_singlesig_p2pkh(
+            StacksPublicKey::from_private(&self.stacks_private_key),
+        )
+        .expect("Failed to create p2pkh spending condition from public key.");
+        sender_spending_condition.set_nonce(sender_nonce);
+
+        let auth = match (payer, payer_nonce) {
+            (Some(payer), Some(payer_nonce)) => {
+                let mut payer_spending_condition =
+                    TransactionSpendingCondition::new_singlesig_p2pkh(
+                        StacksPublicKey::from_private(payer),
+                    )
+                    .expect("Failed to create p2pkh spending condition from public key.");
+                payer_spending_condition.set_nonce(payer_nonce);
+                payer_spending_condition.set_tx_fee(tx_fee);
+                TransactionAuth::Sponsored(sender_spending_condition, payer_spending_condition)
+            }
+            _ => {
+                sender_spending_condition.set_tx_fee(tx_fee);
+                TransactionAuth::Standard(sender_spending_condition)
+            }
+        };
+        let mut unsigned_tx = StacksTransaction::new(self.tx_version, auth, payload);
+        unsigned_tx.anchor_mode = anchor_mode;
+        unsigned_tx.post_condition_mode = TransactionPostConditionMode::Allow;
+        unsigned_tx.chain_id = self.chain_id;
+
+        let mut tx_signer = StacksTransactionSigner::new(&unsigned_tx);
+        tx_signer
+            .sign_origin(&self.stacks_private_key)
+            .map_err(|_| ClientError::SignatureGenerationFailure)?;
+        if let (Some(payer), Some(_)) = (payer, payer_nonce) {
+            tx_signer
+                .sign_sponsor(payer)
+                .map_err(|_| ClientError::SponsorSignatureGenerationFailure)?;
+        }
+
+        let Some(tx) = tx_signer.get_tx() else {
+            return Err(ClientError::SignatureGenerationFailure);
+        };
+
+        Ok(tx.serialize_to_vec())
+    }
+}
+
+impl StacksContractCallable for StacksClient {
+    fn transaction_contract_call(
+        &self,
+        nonce: u64,
+        contract_addr: &StacksAddress,
+        contract_name: &str,
+        function_name: &str,
+        function_args: &[ClarityValue],
+    ) -> Result<Vec<u8>, ClientError> {
+        let contract_name = ContractName::from(contract_name);
```

Comment:
These can panic.  Yes, it's bad rust practice, but it is what it is today.  `::from(x)` just does `::try_from(x).unwrap()` for `ClarityName` and `ContractName`.  This is why this trait interface should use `ContractName` and `ClarityName`.